internal/pkg/configure/tftp.go

```diff
@@ -5,21 +5,22 @@ import (
 	"os"
 	"path"
 
-	"github.com/hpcng/warewulf/internal/pkg/util"
 	warewulfconf "github.com/hpcng/warewulf/internal/pkg/config"
+	"github.com/hpcng/warewulf/internal/pkg/util"
 	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+	"golang.org/x/sys/unix"
 )
 
 func TFTP() error {
 	controller := warewulfconf.Get()
 	var tftpdir string = path.Join(controller.Paths.Tftpdir, "warewulf")
-
+	oldMask := unix.Umask(0)
 	err := os.MkdirAll(tftpdir, 0755)
 	if err != nil {
 		wwlog.Error("%s", err)
 		return err
 	}
-
+	_ = unix.Umask(oldMask)
```

Comment:
I'm noticing that `util.SafeCopyFile` below eventually calls `os.OpenFile`, which is also affected by umask; so restoring the umask here might be premature.

Also, since you can `return err` earlier, you could leak the umask change.

What would you think of just running

```
oldmask := unix.Umask(0)
defer unix.Umask(oldmask)
```

at the top of `TFTP()`?